versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/BaseMergeTransplantSquash.java

```diff
@@ -82,22 +80,27 @@ MergeResult<Commit> squash(
         createSquashCommit(updateCommitMetadata, sourceCommits, mergeFromId);
 
     Map<ContentKey, KeyDetails> keyDetailsMap = new HashMap<>();
+    List<Obj> objsToStore = new ArrayList<>();
     CommitObj mergeCommit =
-        createMergeTransplantCommit(mergeBehaviorForKey, keyDetailsMap, createCommit);
+        createMergeTransplantCommit(mergeBehaviors, keyDetailsMap, createCommit, objsToStore::add);
 
     // It's okay to do the fetchCommit() here and not complicate the surrounding logic (think:
     // local cache)
     StoreIndex<CommitOp> headIndex = indexesLogic(persist).buildCompleteIndexOrEmpty(head);
     verifyMergeTransplantCommitPolicies(headIndex, mergeCommit);
 
+    mergeBehaviors.postValidate();
+
+    // TODO check keyDetailsMap for conflicts here - no need to persist the merge commit
```

Comment:
Good point. We should not call `commitLogic.storeCommit()` if there are conflicts (same for transplanting individual commits).

Reply:
Let's do in a separate PR (same for the comment below)

Reply:
sgtm

Reply:
Yes, I thought so too :)